### PR TITLE
74331 -Functional Tests When AIO Toggle is Off

### PR DESCRIPTION
**Pull request by sheetal13653**
Hi All,

This pull request contains automated FT's with respect to AIO Toggle Off functionality.
[Product Backlog Item 74331](https://dev.azure.com/ukhydro/Exchange%20Set%20Service/_workitems/edit/74331): ESS: Automated Functional Tests when AIO feature is toggle OFF

### CODE REVIEW

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/EssEndPointsScenariosWhenAioIsDisabled.cs [hunk 1]
```diff
@@ -84,5 +84,84 @@ public async Task WhenICallTheProductVersionApiWithValidAndGB800001ProductAndAio
...
+            Assert.AreEqual(200, (int)apiResponse.StatusCode, $"Incorrect status code {apiResponse.StatusCode} is returned, instead of the expected 200.");
+
+            //verify model structure
+            await apiResponse.CheckModelStructureForSuccessResponse();
+
+            var apiResponseData = await apiResponse.ReadAsTypeAsync<ExchangeSetResponseModel>();
+
+            Assert.AreEqual("GB800001", apiResponseData.RequestedProductsNotInExchangeSet.FirstOrDefault().ProductName, $"Exchange set returned Product Name {apiResponseData.RequestedProductsNotInExchangeSet.FirstOrDefault().ProductName}, instead of expected Product Name 'GB800001'");
+            Assert.AreEqual("invalidProduct", apiResponseData.RequestedProductsNotInExchangeSet.FirstOrDefault().Reason, $"Exchange set returned Reason {apiResponseData.RequestedProductsNotInExchangeSet.FirstOrDefault().Reason}, instead of expected Reason 'invalidProduct'");
+
```
Comment (ray-sj) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/EssEndPointsScenariosWhenAioIsDisabled.cs:136]: So the only comment i have is that we should run a test with a mixed bag eg both AIO and other cells. this way only the other cells will be processed and the AIO not.

**Review by ray-sj — CHANGES_REQUESTED** [review]

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/EssEndPointsScenariosWhenAioIsDisabled.cs (hunk 1 above)
Comment (Arit14975) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/EssEndPointsScenariosWhenAioIsDisabled.cs:136]: Hi Ray, We have covered these scenario's . Please check the below file which has all the tests included: 
https://github.com/UKHO/exchange-set-service/blob/Feature/74331-FunctionalTestsWhenAioToggleIsOff/UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/EssEndPointsScenariosWhenAioIsDisabled.cs

**Review by DanOwenUKHO — APPROVED** [review]
LGTM

**Review by Arit14975 — APPROVED** [review]
LGTM. Thanks

**Review by ray-sj — APPROVED** [review]

**Pull request merged**